packages/widgets/src/components/BotsTable/BotsTable.spec.tsx

```diff
@@ -0,0 +1,125 @@
+import { fireEvent, waitFor } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { BotsTableStory } from './stories/BotsTable/StoryComponent';
+import { NoControlOrdersStory } from './stories/NoControlOrders/StoryComponent';
+
+describe('BotsTable', () => {
+  it('Should be correctly rendered', () => {
+    const { getByText, getAllByText } = render(<BotsTableStory />);
+
+    getByText('Bot name 1');
+    getAllByText('27.07.23');
+    getByText('Bittrex');
+    getByText('Pending');
+    getAllByText('ETH/USD');
+    getByText('-70 USD');
+    getByText('-10%');
+  });
+
+  it('Should be alert shown after clicking a table row', () => {
+    const { container } = render(<BotsTableStory />);
+
+    const firstBodyRow = container.querySelector('tbody > tr');
+
+    const alertMock = jest.spyOn(window, 'alert').mockImplementation();
+    fireEvent.click(firstBodyRow as Node);
+    expect(alertMock).toHaveBeenCalledWith(
+      JSON.stringify({
+        id: 1,
+        name: 'Bot name 1',
+        creationDate: '27.07.23',
+        exchange: 'binance',
+        status: 'trade',
+        pair: 'ETH/USD',
+        pnl: { amount: 300, currency: 'USD', isIncreasing: true },
+        roi: { amount: 25, isIncreasing: true },
+        isDisabled: false,
+      }),
+    );
+  });
+
+  it('Should be modal "Run Reason" shown after clicking the icon for "pending" or "liquidation" status', () => {
+    const { container, getByText } = render(<BotsTableStory />);
+
+    const button = container.querySelector(
+      'tbody > tr:nth-child(6) > td:last-child button',
+    );
+
+    fireEvent.click(button as Node);
+
+    getByText('Run Reason');
+  });
+
+  it('Should be submitted "Run Reason" modal', async () => {
+    const consoleMock = jest.spyOn(console, 'log').mockImplementation();
+    const { container, getByText, getByPlaceholderText } = render(
+      <BotsTableStory />,
+    );
+    const button = container.querySelector(
+      'tbody > tr:nth-child(6) > td:last-child button',
+    );
+    fireEvent.click(button as Node);
+    const input = getByPlaceholderText('Enter');
+    fireEvent.change(input, { target: { value: 'reason' } });
+    const runButton = getByText('Run');
+    fireEvent.click(runButton);
+
+    await waitFor(() => {
+      expect(consoleMock).toHaveBeenCalledWith(
+        'Submitting... ' +
+          JSON.stringify({
+            reason: 'reason',
+          }),
+      );
+    });
+  });
+
+  it('Should be clicked by the three dots button', async () => {
```

Comment:
`it should be clicked by the three dots button` means that the table was clicked by the button. I mean, not by a user but by the button, literally.
Use something like this: 
> it should have the menu item clicked

Reply:
Here and in the similar test case below